### PR TITLE
Allow for explain to be set on a query

### PR DESCRIPTION
**Pull request by kyle-mccarthy**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | n/a

Allow for explain to be set on a query, so you can see how the score for a certain document is calculated.

https://www.elastic.co/guide/en/elasticsearch/reference/current/search-explain.html

Can be used by `ElasticActiveRecord::find()->query(...)->explain(true)->all();`.

### CODE REVIEW

File: ActiveRecord.php [hunk 1]
```diff
@@ -193,6 +193,14 @@ public function getHighlight()
     }
 
     /**
+     * @return array|null An explanation for each hit on how its score was computed
+     */
```
Comment (cebe) [inline ActiveRecord.php]: need `@since 2.0.5` annotation.

File: Query.php [hunk 2]
```diff
@@ -171,6 +171,11 @@ class Query extends Component implements QueryInterface
      * @since 2.0.4
      */
     public $options = [];
+    /**
+     * @var Enables explanation for each hit on how its score was computed.
+     * @see https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-explain.html
+     */
```
Comment (cebe) [inline Query.php]: need `@since 2.0.5` annotation.

File: Query.php [hunk 3]
```diff
@@ -688,4 +693,16 @@ public function postFilter($filter)
         $this->postFilter = $filter;
         return $this;
     }
+
+    /**
+     * Explain for how the score of each document was computer
+     * @param $explain
+     * @return $this
+     * @see $explain
+     */
```
Comment (cebe) [inline Query.php]: need `@since 2.0.5` annotation.

**Comment by kyle-mccarthy** [comment]
@cebe Okay I have updated the annotations and added in two simple test cases to check for the existence of `_explanation` on the query when it is set and to ensure it doesn't exist when not set.  Let me know if you see anything else.

**Comment by cebe** [comment]
looks good to me, thanks! If you add a CHANGLOG line I'll merge it right away.

File: Query.php [hunk 4]
```diff
@@ -171,6 +171,12 @@ class Query extends Component implements QueryInterface
      * @since 2.0.4
      */
     public $options = [];
+    /**
+     * @var Enables explanation for each hit on how its score was computed.
```
Comment (cebe) [inline Query.php]: oh, missed that. this should include the type: `@var bool`

File: Query.php [hunk 5]
```diff
@@ -688,4 +694,17 @@ public function postFilter($filter)
         $this->postFilter = $filter;
         return $this;
     }
+
+    /**
+     * Explain for how the score of each document was computer
+     * @param $explain
```
Comment (cebe) [inline Query.php]: same here: `@param bool $explain`

**Comment by kyle-mccarthy** [comment]
Done and not a problem!

**Comment by cebe** [comment]
Merged, thank you!

**Pull request closed without merging**